chain/polkadot/polkadot_chain.go

```diff
@@ -0,0 +1,631 @@
+			VolumeName: v.Name,
+			ImageRef:   chainCfg.Images[0].Ref(),
+			TestName:   testName,
+		}); err != nil {
+			return fmt.Errorf("set volume owner: %w", err)
+		}
+
+		relayChainNodes = append(relayChainNodes, pn)
+	}
+	c.RelayChainNodes = relayChainNodes
+	for _, parachainConfig := range c.parachainConfig {
+		parachainNodes := []*ParachainNode{}
+		for i := 0; i < parachainConfig.NumNodes; i++ {
+			nodeKey, _, err := p2pcrypto.GenerateEd25519Key(crand.Reader)
+			if err != nil {
+				return fmt.Errorf("error generating node key: %w", err)
+			}
+			pn := &ParachainNode{
+				log:             c.log,
+				Index:           i,
+				Chain:           c,
+				DockerClient:    cli,
+				NetworkID:       networkID,
+				TestName:        testName,
+				NodeKey:         nodeKey,
+				Image:           parachainConfig.Image,
+				Bin:             parachainConfig.Bin,
+				ChainID:         parachainConfig.ChainID,
+				Flags:           parachainConfig.Flags,
+				RelayChainFlags: parachainConfig.RelayChainFlags,
+			}
+			v, err := cli.VolumeCreate(ctx, volumetypes.VolumeCreateBody{
+				Labels: map[string]string{
+					dockerutil.CleanupLabel: testName,
+
+					dockerutil.NodeOwnerLabel: pn.Name(),
+				},
+			})
+			if err != nil {
+				return fmt.Errorf("creating volume for chain node: %w", err)
+			}
+			pn.VolumeName = v.Name
+
+			if err := dockerutil.SetVolumeOwner(ctx, dockerutil.VolumeOwnerOptions{
+				Log:        c.log,
+				Client:     cli,
+				VolumeName: v.Name,
+				ImageRef:   parachainConfig.Image.Ref(),
+				TestName:   testName,
+			}); err != nil {
+				return fmt.Errorf("set volume owner: %w", err)
+			}
+			parachainNodes = append(parachainNodes, pn)
+		}
+		c.ParachainNodes = append(c.ParachainNodes, parachainNodes)
+	}
+
+	return nil
+}
+
+func runtimeGenesisPath(path ...interface{}) []interface{} {
+	fullPath := []interface{}{"genesis", "runtime", "runtime_genesis_config"}
+	fullPath = append(fullPath, path...)
+	return fullPath
+}
+
+func (c *PolkadotChain) modifyGenesis(ctx context.Context, chainSpec interface{}) error {
+	bootNodes := []string{}
+	authorities := [][]interface{}{}
+	balances := [][]interface{}{}
+	var sudoAddress string
+	for i, n := range c.RelayChainNodes {
+		multiAddress, err := n.MultiAddress()
+		if err != nil {
+			return err
+		}
+		bootNodes = append(bootNodes, multiAddress)
+		stashAddress, err := n.StashAddress()
+		if err != nil {
+			return fmt.Errorf("error getting stash address")
+		}
+		accountAddress, err := n.AccountAddress()
+		if err != nil {
+			return fmt.Errorf("error getting account address")
+		}
+		grandpaAddress, err := n.GrandpaAddress()
+		if err != nil {
+			return fmt.Errorf("error getting grandpa address")
+		}
+		beefyAddress, err := n.EcdsaAddress()
+		if err != nil {
+			return fmt.Errorf("error getting beefy address")
+		}
+		balances = append(balances,
+			[]interface{}{stashAddress, 1000000000000000000},
+			[]interface{}{accountAddress, 1000000000000000000},
```

Comment:
I noticed this [build error](https://github.com/cosmos/ibc-go/runs/7956132181?check_suite_focus=true) for arm machines:
```
Error: ../../../../go/pkg/mod/github.com/strangelove-ventures/ibctest@v0.0.0-20220822145344-eb5511e535a8/chain/polkadot/polkadot_chain.go:285:32: cannot use 1000000000000000000 (untyped int constant) as int value in array or slice literal (overflows)
```